docs/contributing/devguide.md

```diff
@@ -46,6 +52,104 @@ sidebar_position: 3
+    - 80 characters is the recommended column
+- [GitLens](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens)
+    - Hover over lines to see commit history, last author
+- [Remote Development](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack)
+    - For working over SSH
+
+## :construction: Git Guidelines
+
+### :deciduous_tree: Branches
+
+The branch name should be copied from a [Arrow
+DeWork](https://app.dework.xyz/arrow-air) task.
+
+![DeWork - Suggested Git Branch Name](./assets/dework_branch.png)
+- e.g. `am-smith/dw-227/rust-service-template`
+- This links your work to a ticket
+- Closing a pull request with this branch will also mark the DeWork task as "Done"!
+
+### :lock_with_ink_pen: Sign Your Commits
+
+We enforce signed commits to authenticate commits from an author.
+- Security, accountability, traceability
+- ["How (and why) to sign Git commits" - Alessandro Segala](https://withblue.ink/2020/05/17/how-and-why-to-sign-git-commits.html)
+
+:exclamation: All remote branches are protected from unsigned commits through
+GitHub repository settings.
+
+Setting up a GPG Key:
+- [Generate a GPG
+  Key](https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key)
+- [Informing Git about your GPG Key](https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key)
+- [Informing GitHub about your GPG Key](https://docs.github.com/en/authentication/managing-commit-signature-verification/adding-a-new-gpg-key-to-your-github-account)
+
+Add an `-S` to your commit command to sign it, e.g. `git commit -S -m "Add
+README.md"`.
+
+### :speech_balloon: Commit Messages
+
+As described [here](https://github.com/marketplace/actions/dynamic-badges#git-commit-messages):
+- Message should be in present tense (**Add new feature** instead of **Added new feature**)
+  - Should be written as if giving a command or instruction (**Remove bug X**, **Update README.md**)
+- Message should start with a capital letter (**Add new feature**)
+- Emojis, if used, should be at the start of the message (**:tada: Add new feature**)
+
+Additionally:
+- Do not exceed a line length of 50 characters
+- Do **not** end the message with a period
+
+Keep these in mind always, especially when squashing commits (one message).
+
+### :+1: Pull Requests
+
+- Keep PRs small and easy to review!
+    - Our [reviewers](./peerreview.md) may ask you to divide your PR into
+      multiple smaller PRs.
+- If the commit history is messy, recommend "Squash and Merge" when pushing in
+  changes.
+    - The squashed commit should have a thorough description of the changes
+      added.
+- The Pull Request comments should be descriptive
+    - A brief description of what was changed and why
+- Your PR should equate to one DeWork issue
+    - Addressing multiple tickets in one merge makes it harder to debug breaking changes.
```

Comment:
```suggestion
    - Addressing multiple tickets in one merge makes it harder to debug breaking changes.
    - Use `git commit --fixup <commit sha>` to fix a previous commit. 
```

Reply:
We should configure a GitHub action that allows autosquash by simply commenting certain keywords:
https://github.com/marketplace/actions/automatic-rebase